src/Interpreters/TemporaryDataOnDisk.h

```diff
@@ -90,14 +90,20 @@ class TemporaryDataOnDisk : private TemporaryDataOnDiskScope
     /// If max_file_size > 0, then check that there's enough space on the disk and throw an exception in case of lack of free space
     TemporaryFileStream & createStream(const Block & header, size_t max_file_size = 0);
 
+    /// Write raw data directly into buffer.
+    /// Differences from `createStream`:
+    ///   1) it doesn't account data in parent scope
+    ///   2) returned buffer owns resources (instead of TemporaryDataOnDisk itself)
+    WriteBufferPtr createRawStream(size_t max_file_size = 0);
```

Comment:
```suggestion
    /// If max_file_size > 0, then check that there's enough space on the disk and throw an exception in case of lack of free space
    WriteBufferPtr createRawStream(size_t max_file_size = 0);
```